### PR TITLE
gradient support!

### PR DESCRIPTION
**Pull request by ameliasquires**
added `fx_render_pass_add_rect_grad`, `fx_render_pass_add_rounded_rect_grad`, and `fx_render_pass_add_rounded_grad_border_corner`. all with associated shaders, structs, and linking functions.

all gradients support as many colors as you want (dynamic shader compilation), linear/conic variants, and rotation (in degrees)

### CODE REVIEW

**Comment by WillPower3309** [comment]
This is huge! Thanks a ton for the PR. Feel free to ping me directly when it's ready for review

File: include/scenefx/render/pass.h [hunk 1]
```diff
@@ -27,6 +27,20 @@ struct fx_gles_render_pass *fx_renderer_begin_buffer_pass(struct wlr_renderer *w
 		struct wlr_buffer *wlr_buffer, struct wlr_output *output,
 		const struct wlr_buffer_pass_options *options);
 
+struct fx_gradient {
+	float degree;
```
Comment (WillPower3309) [inline include/scenefx/render/pass.h:30]: What are the bounds of this?

File: render/fx_renderer/gles2/shaders/gradient.frag [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+vec4 gradient(vec4 colors[LEN], int count, vec2 size, vec2 grad_box, vec2 origin, float degree, bool linear, bool blend){
+	float step;
+
+	vec2 normal = (gl_FragCoord.xy - grad_box)/size;
+	vec2 uv = normal - origin;
+
+	float rad = radians(degree);
+
+	if(linear){
```
Comment (WillPower3309) [inline render/fx_renderer/gles2/shaders/gradient.frag]: Nit: please match the bracket spacing throughout the shaders here with the existing format we have in our other shaders

**Review by WillPower3309 — COMMENTED** [review]
One Q and one nit. Otherwise looks great! Clean PR

**Review by WillPower3309 — APPROVED** [review]

**Comment by IDKnix** [comment]
lets go it got merged

**Pull request merged**